packages/seacas/libraries/ioss/src/Ioss_DynamicTopology.C

```diff
@@ -36,14 +36,463 @@
+    if(comparator(stepTimeDiff, minTimeDiff)) {
+      minTimeDiff = stepTimeDiff;
+      loc.time  = stateTime;
+      loc.state = istep;
+      loc.changeSet = db->supports_change_set() ? db->get_change_set_name() : db->get_filename();
+    }
+  }
+}
+
+void locate_state(Ioss::DatabaseIO *db, double targetTime, DatabaseState& loc)
+{
+  // Get state count and all states...
+  std::vector<double> timesteps = db->get_db_step_times();
+  int stepCount = timesteps.size();
+
+  if(targetTime < 0.0) {
+    // Round towards 0
+    StateLocatorCompare compare = [](double a, double b) { return (a <= b); };
+    locate_state_impl(db, targetTime, compare, loc);
+  }
+  else {
+    // Round towards 0
+    StateLocatorCompare compare = [](double a, double b) { return (a < b); };
+    locate_state_impl(db, targetTime, compare, loc);
+  }
+}
+
+DatabaseState locate_group_db_state(Ioss::DatabaseIO *db, double targetTime)
+{
+  DatabaseState loc(db);
+  std::string currentChangeSet = db->get_change_set_name();
+
+  for(int i=0; i<db->num_change_set(); i++) {
+    db->open_change_set(i);
+    locate_state(db, targetTime, loc);
+  }
+
+  db->open_change_set(currentChangeSet);
+
+  return loc;
+}
+
+bool file_exists(const Ioss::ParallelUtils &util,
+                 const std::string &filename,
+                 const std::string &db_type,
+                 Ioss::DatabaseUsage db_usage)
+{
+  bool exists = false;
+  int par_size = util.parallel_size();
+  int par_rank = util.parallel_rank();
+  bool is_parallel = par_size > 1;
+  std::string full_filename = filename;
+  if (is_parallel && db_type == "exodusII" && db_usage != Ioss::WRITE_HISTORY) {
+    full_filename = Ioss::Utils::decode_filename(filename, par_rank, par_size);
+  }
+
+  if (!is_parallel || par_rank == 0) {
+    // Now, see if this file exists...
+    // Don't want to do a system call on all processors since it can take minutes
+    // on some of the larger machines, filesystems, and processor counts...
+    Ioss::FileInfo file = Ioss::FileInfo(full_filename);
+    exists = file.exists();
+  }
+
+  if (is_parallel) {
+    int iexists = exists ? 1 : 0;
+    util.broadcast(iexists, 0);
+    exists = iexists == 1;
+  }
+  return exists;
+}
+
+int file_exists( const Ioss::ParallelUtils &util,
+                 const std::string& filename,
+                 std::string& message,
+                 bool specifiedDecomp )
+{
+  std::string filenameBase = filename;
+  const int par_size = util.parallel_size();
+  const int par_rank = util.parallel_rank();
+
+  if( par_size > 1 && !specifiedDecomp ) {
+    filenameBase = Ioss::Utils::decode_filename(filenameBase, par_rank, par_size);
+  }
+
+  Ioss::FileInfo file = Ioss::FileInfo(filenameBase);
+  return file.parallel_exists(util.communicator(), message);
+}
+
+bool internal_decomp_specified(const Ioss::PropertyManager& props)
+{
+  bool internalDecompSpecified = false;
+
+  const std::string restartDecompMethod("RESTART_DECOMPOSITION_METHOD");
```

Comment:
This could also be `DECOMPOSITION_METHOD`

But I'm not sure this is robust since we now have similar code in multiple places -- exodus IOFactory and here and maybe other places.

Reply:
So should I check for both strings or have some encapsulation of this in Ioss::Utils ?